### PR TITLE
Flap damp scoring

### PR DESCRIPTION
**Pull request by jwolski**
This is the first of several PRs to come for flap damping. The first is damp scoring. A member in the membership list will be penalized for every update that is applied. When the damp score exceeds the configured damp suppress limit, an event will be emitted. Later, this event will initiate the damping subprotocol. (That's the next PR to come). The damp scores for all members decay exponentially, its periodicity governed by another configurable parameter. 

I left a few cheeky comments and added a few bells/whistles (like the separate Config object) to this PR to accompany the scoring. Don't mind them too much.

The test coverage is fairly reasonable.

Oh, and also... tick-cluster needed a bit of fixing. A member's damp score is now part of the returned stats. The way tick-cluster previously computed checksums for membership was too restrictive.

There's actually no net gain in terms of functionality in this PR. It's simply a mechanism to start scoring members and learning about scoring, in general.

@danielheller @benfleis @CorgiMan

### CODE REVIEW

File: config.js [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+util.inherits(Config, EventEmitter);
+
+Config.prototype.get = function get(key) {
+    return this.store[key];
+};
+
+Config.prototype.set = function set(key, value) {
+    var oldValue = this.store[key];
+    this.store[key] = value;
+    this.emit('changed', key, value, oldValue);
```
Comment (benfleis) [inline config.js]: perhaps 'update' or 'set', since those are the action labels we use elsewhere?

File: lib/membership/member.js [hunk 2]
```diff
@@ -28,10 +28,39 @@ function Member(ringpop, update) {
     this.address = update.address;
     this.status = update.status;
     this.incarnationNumber = update.incarnationNumber;
+    this.dampScore = update.dampScore || ringpop.config.get('dampScoringInitial');
```
Comment (benfleis) [inline lib/membership/member.js]: can update.dampScore === 0?

File: lib/membership/member.js [hunk 3]
```diff
@@ -59,8 +88,24 @@ Member.prototype.evaluateUpdate = function evaluateUpdate(update) {
...
+        // will be penalized. The scoring levers will control persistent
+        // flaps. We'll eventually get _real_ good at identifying flaps
+        // and apply penalties more strictly.
+        this._applyUpdatePenalty();
+        this.lastUpdateDampScore = this.dampScore;
+    }
+
     this.emit('updated', update);
 
+    // lastUpdateTimestamp must be updated after the penalty is applied
```
Comment (benfleis) [inline lib/membership/member.js]: I <3 this kind of comment.

File: test/unit/config_test.js [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+    assert.plan(2);
+
+    var config = new Config();
+    config.on('changed', onConfigChanged);
+    var configKey = 'somekey';
+    config.on('changed.' + configKey, onConfigKeyChanged);
+    var configVal = 'someval';
+    config.set(configKey, configVal); // this triggers change events *fingers crossed*
+
+    assert.end();
```
Comment (benfleis) [inline test/unit/config_test.js]: Maybe I misunderstand .pass(), but I don't think this test can fail? would it be better to set count = 0, and have each event callback incr, and assert count == 2?

Comment (benfleis) [inline test/unit/config_test.js]: Ahh, I see. .plan tells it to require 2 asserts. Ignore me, since .plan + .pass does exactly what I described :)

**Comment by benfleis** [comment]
done with this round. only potential bug is the explicit == 0 possibility above. other than that and 'changed', lgtm

File: config.js (hunk 1 above)
Comment (jwolski) [inline config.js]: yea, i think that's a good suggestion. ill "change" it ;)

File: lib/membership/member.js (hunk 2 above)
Comment (jwolski) [inline lib/membership/member.js]: yes, good catch. Since dampScores are not yet disseminated, this isn't an issue...yet. I'll fix no matter.

**Pull request merged**